src/services/consent/services/consentVersionService.js

```diff
@@ -8,19 +8,18 @@ const {
 
 const {
 	restrictToCurrentSchool,
+	denyIfNotCurrentSchoolOrEmpty,
 	hasPermission,
 } = require('../../../hooks');
 
 const { modelServices: { prepareInternalParams } } = require('../../../utils');
 
 const ConsentVersionServiceHooks = {
 	before: {
-		all: [],
+		all: [authenticate('jwt'), iff(isProvider('external'), restrictToCurrentSchool)],
```

Comment:
restrictToCurrentSchool can not work for us.

Only in one instance we activated this feature for schools. For all others instances we have consentVersions from us for the complete instance. We also use this point for the instance with this feature, to post new consents from our side maybe a new messenger is go on.

We must write a own hooks that first search if a consentVersion for the school exist. If it exist it must get access this this consentVersion and "i think" also for all without schoolId. I think we can force this over a query schoolId or "not set" | undefined | null. 

The second is, a not superhero should only upload new consentVersion files if this feature is enabled and only for this own school. Maybe force schoolId in the query for this condition. Otherwise no other non superhero should upload / create anything. 
In admin overview it should only display the consents with school. You can force this if you set schoolId in request and if not it is displayed all with schoolId and with undefined. 

Maybe you must also check how it is implement in client and some points. 

Sorry for re-think the process and search all related points. But at the moment it can not work and it must work for all related task. 

Please add also test that clarify what must work and what not.

Reply:
I see line 81 already force some points.

Reply:
Hi, I am trying to understand what should i change in the code we talked about yesterday. From your comment I was able to understand the following:
1. Access to consent version (I assume that it is some document) should be available for everyone event if they dont have schoolId assigned (determined only by condition if school has such consent doc)
2. Write a hook which will only allow pushing new consent version for a superhero user.

I don't get this one: "In admin overview it should only display the consents with school. You can force this if you set schoolId in request and if not it is displayed all with schoolId and with undefined."

Reply:
@CeEv I think you're missing that there are two separate services for the two use cases now (see RegistrationConsentService below). The logic that you refer to is in the find method (where it should be, not in the hook)

Reply:
What we finally need, as i know and it does not matter for me if this is implemented via hook (but i prefere) or in the service is:
- have consentVersions displayed in school admin area and for firstlogin and (datasecurity page or footer)
- read access is forbidden if a foreign schoolId is given (filter own or no school id)
- write access is given iff (create & schoolId given) or update and (own schoolId and not changing schoolId) but not without school id

Reply:
All right but RegistrationConsentService is also forced the schoolId and restrictToCurrentSchool is force it for this service. 
In the end we have the situation that we can not push non school consentVersions from us to the users that they can read and accept it.
My wish is that the API and the client are checked and that the tests also map this.

Reply:
Can you write in points which checks needs to be performed? I am now pretty confused :/

Reply:
@CeEv @Parnswir @tofixx Up of this topic. As I mentioned in above comment, can you provide some common point about this PR (what needs to be done if anything at all)?